apps/etterlatte-behandling/src/main/kotlin/institusjonsopphold/InstitusjonsoppholdRoutes.kt

```diff
@@ -5,20 +5,35 @@ import io.ktor.server.application.call
 import io.ktor.server.request.receive
 import io.ktor.server.response.respond
 import io.ktor.server.routing.Route
+import io.ktor.server.routing.get
 import io.ktor.server.routing.post
 import io.ktor.server.routing.route
 import no.nav.etterlatte.libs.common.SAKID_CALL_PARAMETER
+import no.nav.etterlatte.libs.common.grunnlag.Grunnlagsopplysning
+import no.nav.etterlatte.libs.common.hentNavidentFraToken
 import no.nav.etterlatte.libs.common.sakId
 
 internal fun Route.institusjonsoppholdRoute(institusjonsoppholdService: InstitusjonsoppholdService) {
     route("/api/institusjonsoppholdbegrunnelse/{$SAKID_CALL_PARAMETER}") {
         post {
-            val institusjonsoppholdBegrunnelse = call.receive<InstitusjonsoppholdBegrunnelseWrapper>()
-            institusjonsoppholdService.leggInnInstitusjonsoppholdBegrunnelse(
-                sakId,
-                institusjonsoppholdBegrunnelse.institusjonsopphold
+            hentNavidentFraToken { navIdent ->
+                val institusjonsoppholdBegrunnelse = call.receive<InstitusjonsoppholdBegrunnelseWrapper>()
+                institusjonsoppholdService.leggInnInstitusjonsoppholdBegrunnelse(
+                    sakId,
+                    Grunnlagsopplysning.Saksbehandler.create(navIdent),
+                    institusjonsoppholdBegrunnelse.institusjonsopphold
+                )
+                call.respond(HttpStatusCode.OK)
+            }
+        }
+    }
+
+    route("/api/institusjonsoppholdbegrunnelse/{grunnlagsendringsid}") {
+        get {
+            val tmp = institusjonsoppholdService.hentInstitusjonsoppholdBegrunnelse(
+                call.parameters["grunnlagsendringsid"]!!
             )
-            call.respond(HttpStatusCode.OK)
+            call.respond(tmp ?: HttpStatusCode.NotFound)
```

Comment:
Bør denne vera NotFound eller NoContent? 🤔

Reply:
Jeg prøver å vise data basert på at requesten er success så her passer det best med not found slik at ikke frontend tryner :D

Reply:
Ah, så viss det ikkje fins noko her så er det faktisk ein feil, så 404 er greitt. Got it 💡

Reply:
Jepp, fremover vil det være slik!